functional_testing/Openvcloud/ovc_master_hosted/OVC/c_advanced/machine_tests.py

```diff
@@ -744,11 +744,52 @@ def test015_check_IP_conflict(self):
         cs_ip = self.api.cloudapi.cloudspaces.get(self.cloudspace_id)["publicipaddress"]
 
         vm_1_ext_client.execute("ip a d {} dev {}".format(vm1_ext_ip,ext_interface_name1), sudo=True)
-        vm_1_client = VMClient(vm1_id)        
+        vm_1_client = VMClient(vm1_id)
         vm_1_client.execute("ip a a {} dev {}".format(cs_ip,ext_interface_name1), sudo=True)
         vm_1_client.execute("nohup bash -c 'ip l s dev {} up </dev/null >/dev/null 2>&1 & '".format(ext_interface_name1), sudo=True)
 
         self.lg("Check that you can't connect to VM1 with it's new ext network ip, should succeed")
         with self.assertRaises(socket.error):
             VMClient(vm1_id,ip=cs_ip, external_network=True, timeout=1)
         self.lg('%s ENDED' % self._testID)
+
+    def test016_resize_disk_online(self):
+        """ OVC-047
+        *Test case for resizing disk online*
+        **Test Scenario:**
+
+        #. Create virtual machine (VM1), should succeeed
+        #. Attach data disk (DS1) with size (S1), should succeed
+        #. Resize DS1 to S2, should succeed
+        #. Make sure DS1's size has been changed on the vm itself
+        #. Resize VM1's metadata disk, should fail
+        """
+        self.lg('%s STARTED' % self._testID)
+
+        self.lg('Create virtual machine (VM1), should succeeed')
+        vm1_id = self.cloudapi_create_machine(cloudspace_id=self.cloudspace_id)
+
+        self.lg('Attach data disk (DS1) with size (S1), should succeed')
+        s1 = random.randint(10, 50)
+        disk_id = self.create_disk(self.account_id, size=s1)
+        self.assertTrue(disk_id)
+        response = self.api.cloudapi.machines.attachDisk(machineId=vm1_id, diskId=disk_id)
+        self.assertTrue(response)
+
+        self.lg('Resize DS1 to S2, should succeed')
+        self.api.cloudapi.disks.resize(diskId=disk_id, size=s1 + 1)
+
```

Comment:
add step to try to increase disk size and it should fail

Reply:
no need .. this api is tested before